secure/storage/src/in_memory.rs

```diff
@@ -70,6 +70,7 @@ impl<T: Send + Sync + TimeService> KVStorage for InMemoryStorageInternal<T> {
                 Value::Ed25519PrivateKey(key)
             }
             Value::HashValue(value) => Value::HashValue(*value),
+            Value::String(value) => Value::String(value.clone()),
```

Comment:
FYI: It would be nice if we didn't have to update this hack for every new Value type.. I might look into this if I have a few cycles.. but for now it's probably fine!

Reply:
I started hacking on this... the easy way to do it is is to make the set, create, get generic with respect to some serializable type. Unfortunately generic functions are incompatible with trait objects. So in order to get this to work effectively (using this solution), we'd need to eliminate our trait objects on storage and use templates throughout.

Agreed, it is bad and it requires storing metadata with data when storing and loading...


{"type": "u64", "value": 0} could just be 0